src/common/hooks/useCancelOrder/useGetOnChainCancellation.ts

```diff
@@ -14,17 +15,24 @@ import {
 export function useGetOnChainCancellation(): (order: Order) => Promise<OnChainCancellation> {
   const ethFlowContract = useEthFlowContract()
   const settlementContract = useGP2SettlementContract()
+  const cancelTwapOrder = useCancelTwapOrder()
 
   return useCallback(
     (order: Order) => {
+      const composableCowId = order.composableCowInfo?.id
+
       const isEthFlowOrder = getIsEthFlowOrder(order)
 
+      if (composableCowId) {
+        return cancelTwapOrder(composableCowId)
+      }
```

Comment:
You can move the condition above if it takes precedence over ethflow order cancellation:

```suggestion
      if (composableCowId) {
        return cancelTwapOrder(composableCowId)
      }
      
      const isEthFlowOrder = getIsEthFlowOrder(order)
```

Reply:
Thanks! Fixed